drv/gimlet-seq-server/src/main.rs

```diff
@@ -1146,9 +1178,16 @@ fn read_spd_data_and_load_packrat(
             let spd = I2cDevice::new(i2c_task, controller, port, mux, mem);
 
             let chunk = 128;
-            retry_i2c_txn(|| spd.read_reg_into::<u8>(0, &mut tmp[..chunk]))?;
 
-            retry_i2c_txn(|| spd.read_into(&mut tmp[chunk..]))?;
+            retry_i2c_txn(I2cTxn::SpdLoadTop(nbank, i), || {
+                //
+                // Both of these reads need to be in a single transaction from
+                // the perspective of the retry logic: if either fails, we
+                // must redo both.
+                //
+                spd.read_reg_into::<u8>(0, &mut tmp[..chunk])?;
+                spd.read_into(&mut tmp[chunk..])
```

Comment:
Dumb question: why do these reads need to be separate, versus
```suggestion
                spd.read_reg_into::<u8>(0, &mut tmp)?;
```

Reply:
Same reply here: the 255 byte limit on i2c operations.